app/controllers/api/base_controller/action.rb

```diff
@@ -19,21 +19,21 @@ def api_action(type, id, options = {})
       # - constructs action_result for successes and failures
       # - throws errors for single resources and use results for multiple resoruces
       def api_resource(type, id, action_phrase)
-        id ||= @req.collection_id
-        klass = collection_class(type)
-        raise BadRequestError, "#{action_phrase} #{type.to_s.titleize} requires an id" unless id
+        api_action(type, id) do |klass|
+          id ||= @req.collection_id
+          raise BadRequestError, "#{action_phrase} #{type.to_s.titleize} requires an id" unless id
 
-        api_log_info("#{action_phrase} #{type.to_s.titleize} id: #{id}")
-        resource = resource_search(id, type, klass)
-        result_options = yield(resource)
-        result = action_result(true, "#{action_phrase} #{model_ident(resource, type)}", result_options)
-        add_href_to_result(result, type, id)
-        log_result(result)
-        result
-      rescue ActiveRecord::RecordNotFound, ForbiddenError, BadRequestError, NotFoundError => err
-        single_resource? ? raise : action_result(false, err.to_s)
-      rescue => err
-        action_result(false, err.to_s)
+          api_log_info("#{action_phrase} #{type.to_s.titleize} id: #{id}")
+          resource = resource_search(id, type, klass)
+          result_options = yield(resource)
+          # result_options is either action_hash options (preferred) or full action hash (finer grained messaging)
+          result = action_result(true, "#{action_phrase} #{model_ident(resource, type)}", result_options) unless result_options.key?(:success)
+          result
```

Comment:
what happens to result in the opposite case (when result_options does have a :success key).  I think it would be nil, and then what does the caller do?

Reply:
Should this actually return result_options, since that's what's returned form the yield?

Reply:
good catch

Reply:
we don't call this new code path yet, that is how it slipped through tests